dev/generate-kernel-signatures.py

```diff
@@ -429,7 +429,10 @@ def by_signature(cuda_kernel_templates):
                 special = [repr(spec["name"])]
                 [type_to_pytype(x["type"], special) for x in childfunc["args"]]
                 dirlist = [repr(x["dir"]) for x in childfunc["args"]]
-                ispointerlist = [repr("List" in x["type"]) for x in childfunc["args"]]
+                ispointerlist = [
+                    repr("List" in x["type"] or "ListArray-at" == x.get("role", None))
```

Comment:
This is a first use of 'role' to specify that the type is a pointer in a cuda kernel. There are three kernels that use this role: two of them are in this PR and the third one is `awkward_ListArray_getitem_jagged_expand`.

Reply:
Right now, the roles are like untested documentation: I don't think anything relies on them (except this `ispointerlist`, now), so they might be inaccurate or not correspond to the `ispointerlist` condition that you want here.

Reply:
The `ListArray-at` role is not defined nor used anywhere else. This is it's first usage definition. Let's discuss it on Wednesday.